File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayMoneyValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayMoneyInValue.OptArrayMoneyValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayMoneyValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayMoneyValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                java.math.BigDecimal[].class,
                OptArrayMoneyValue.class.getDeclaredField("optArrayMoney").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                java.math.BigDecimal[].class,
                OptArrayMoneyValue.class.getMethod("getOptArrayMoney").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayMoneyValue.class,
                OptArrayMoneyValue.class.getMethod("setOptArrayMoney", java.math.BigDecimal[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayMoneyValue().getOptArrayMoney());
    }
}

File: legacy/workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayMoneyValue.scala
package com.dslplatform.ocd.values

import OptArrayMoneyInValue.OptArrayMoneyValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayMoneyValue extends SpecExtensions {
  def `Test property field type: Option[Array[BigDecimal]]`() =
    checkType(
      typeOf[Option[Array[BigDecimal]]]
    , typeOf[OptArrayMoneyValue].member("optArrayMoney": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayMoneyValue().optArrayMoney === None)
  }
}
